### PR TITLE
Fix issue where we weren't properly inferring a delegate type in a signature.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This was causing generate-method to not offer a proper fix when it should have been.

Fixes https://github.com/dotnet/roslyn/issues/8230

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Workspaces/Core/Portable/Shared/Extensions/ITypeInferenceServiceExtensions.cs [hunk 1]
```diff
@@ -16,8 +16,9 @@ internal static class ITypeInferenceServiceExtensions
             SyntaxNode expression,
             CancellationToken cancellationToken)
         {
-            var type = typeInferenceService.InferType(semanticModel, expression, objectAsDefault: false, cancellationToken: cancellationToken);
-            return type.GetDelegateType(semanticModel.Compilation);
+            var types = typeInferenceService.InferTypes(semanticModel, expression, cancellationToken);
+            var delegateTypes = types.Select(t => t?.GetDelegateType(semanticModel.Compilation));
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/Shared/Extensions/ITypeInferenceServiceExtensions.cs]: Where would `InferTypes` return `null`s, such that you need `t?`?

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/ITypeInferenceServiceExtensions.cs]: It wouldn't. I'll fix up.

**Comment by Pilchie** [comment]
:+1:

**Comment by jasonmalinowski** [comment]
:+1:

**Pull request merged**